detect_secrets/plugins/common/util.py

```diff
@@ -3,31 +3,55 @@
 except ImportError:  # pragma: no cover
     from functools32 import lru_cache
 
-# These plugins need to be imported here so that globals()
-# can find them.
-from ..artifactory import ArtifactoryDetector               # noqa: F401
-from ..aws import AWSKeyDetector                            # noqa: F401
-from ..base import BasePlugin
-from ..basic_auth import BasicAuthDetector                  # noqa: F401
-from ..high_entropy_strings import Base64HighEntropyString  # noqa: F401
-from ..high_entropy_strings import HexHighEntropyString     # noqa: F401
-from ..jwt import JwtTokenDetector                          # noqa: F401
-from ..keyword import KeywordDetector                       # noqa: F401
-from ..mailchimp import MailchimpDetector                   # noqa: F401
-from ..private_key import PrivateKeyDetector                # noqa: F401
-from ..slack import SlackDetector                           # noqa: F401
-from ..stripe import StripeDetector                         # noqa: F401
+import os
+from abc import abstractproperty
+from importlib import import_module
+
+from detect_secrets.plugins.base import BasePlugin
+from detect_secrets.util import get_root_directory
 
 
 @lru_cache(maxsize=1)
 def get_mapping_from_secret_type_to_class_name():
     """Returns secret_type => plugin classname"""
-    mapping = {}
-    for key, value in globals().items():
-        try:
-            if issubclass(value, BasePlugin) and value != BasePlugin:
-                mapping[value.secret_type] = key
-        except TypeError:
-            pass
-
-    return mapping
+    return {
+        value.secret_type: key
+        for key, value in import_plugins().items()
```

Comment:
Super nit: I know it says plugin classname above, but I liked when you did e.g.
```
for name, plugin in import_plugins().items()
```
in other places, as it was more immediately clear. / Consistency nit.